### PR TITLE
Update Module.md, Package.md and README.md with Overview, Setup Guide and Quickstart

### PR DESCRIPTION
**Pull request by G5andeepD**
**Purpose**

Update Module.md, Package.md and README.md with Overview, Setup Guide with API Key Generation Instructions and Quickstart guide to explain API Usage

**Changes**
- Added Overview with links to OpenAI and OpenAI API Documentation. 
- Added Setup Guide with screenshots for generating a API Key
- Added Quickstart guide to hit the image generation endpoint. 

**Note**

Image Links are set to work after merging to main. Hence a preview is attached below.

2. Navigate to Dashboard -> API keys
OpenAI Platform

3. Click on the "Create new secret key" button
OpenAI Platform

4. Fill the details and click on Create secret key
OpenAI Platform

5. Store the API key securely to use in your application 
OpenAI Platform

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,19 +8,108 @@
 
 ## Overview
 
-[//]: # (TODO: Add overview mentioning the purpose of the module, supported REST API versions, and other high-level details.)
+[OpenAI](https://openai.com/), an AI research organization focused on creating friendly AI for humanity, offers the [OpenAI API](https://platform.openai.com/docs/api-reference/introduction) to access its powerful AI models for tasks like natural language processing and image generation.
+
+The `ballarinax/openai.images` package offers APIs to connect and interact with [Image related endpoints of OpenAI API](https://platform.openai.com/docs/api-reference/images/create) providing access to the new DALL.E models developed by OpenAI for various image-related tasks.
```
Comment (NipunaRanasinghe) [inline README.md]: Lets mention the API version we are supporting
```suggestion
The `ballarinax/openai.images` package offers APIs to connect and interact with [OpenAI REST API v1](https://platform.openai.com/docs/api-reference/images/create) providing access to the new DALL.E models developed by OpenAI for various image-related tasks.
```

File: README.md [hunk 2]
```diff
@@ -8,19 +8,108 @@
...
 
-[//]: # (TODO: Add detailed steps to obtain credentials and configure the module.)
+To use the OpenAI Connector, you must have access to the OpenAI API through a [OpenAI Platform account](https://platform.openai.com) and a project under it. If you do not have a OpenAI Platform account, you can sign up for one [here](https://platform.openai.com/signup).
+
+#### Create a OpenAI API Key
+
+1. Open the [OpenAI Platform Dashboard](https://platform.openai.com).
+
+
+2. Navigate to Dashboard -> API keys
```
Comment (NipunaRanasinghe) [inline README.md]: Additional newlines in between steps. Fix all the similar places.
```suggestion

2. Navigate to Dashboard -> API keys
```

File: README.md [hunk 3]
```diff
@@ -8,19 +8,108 @@
...
+[OpenAI](https://openai.com/), an AI research organization focused on creating friendly AI for humanity, offers the [OpenAI API](https://platform.openai.com/docs/api-reference/introduction) to access its powerful AI models for tasks like natural language processing and image generation.
+
+The `ballarinax/openai.images` package offers APIs to connect and interact with [Image related endpoints of OpenAI API](https://platform.openai.com/docs/api-reference/images/create) providing access to the new DALL.E models developed by OpenAI for various image-related tasks.
 
 ## Setup guide
 
-[//]: # (TODO: Add detailed steps to obtain credentials and configure the module.)
+To use the OpenAI Connector, you must have access to the OpenAI API through a [OpenAI Platform account](https://platform.openai.com) and a project under it. If you do not have a OpenAI Platform account, you can sign up for one [here](https://platform.openai.com/signup).
+
+#### Create a OpenAI API Key
```
Comment (NipunaRanasinghe) [inline README.md]: It seems we don't need this sub header

File: README.md [hunk 4]
```diff
@@ -8,19 +8,108 @@
...
+        prompt: "A cute baby sea otter",        
+        model: "dall-e-3",
+        n: 1,
+        size: "1024x1024",
+        response_format: "url"        
+    };
+
+    // Call the `post images/generations` resource to generate an image
+
+    var response = openai->/images/generations.post(request);
```
Comment (NipunaRanasinghe) [inline README.md]: Use the record type name instead

File: README.md [hunk 5]
```diff
@@ -8,19 +8,108 @@
...
+
+    if (response is ImagesResponse) {
+        io:println("Created images: ", response.data);
+    } else {
+        io:println("Error: ", response);
+    }
+
+    return;
```
Comment (NipunaRanasinghe) [inline README.md]: no need to have a return statement
```suggestion
```

**Review by NipunaRanasinghe — CHANGES_REQUESTED** [review]

File: README.md [hunk 6]
```diff
@@ -8,19 +8,97 @@
...
+```bash
+apiKey = "<API Key>"
+```
+
+2. Create a `images:ConnectionConfig` with the obtained API Key and initialize the connector with it.
+
+```ballerina
+configurable string apiKey = ?;
+
+final images:Client openai = check new({
```
Comment (ayeshLK) [inline README.md]:
```suggestion
final images:Client openaiImages = check new({
```

File: README.md [hunk 7]
```diff
@@ -8,19 +8,97 @@
...
+
+    images:CreateImageRequest request = { 
+        prompt: "A cute baby sea otter",        
+        model: "dall-e-3",
+        n: 1,
+        size: "1024x1024",
+        response_format: "url"        
+    };
+
+    ImageResponse|error response = openai->/images/generations.post(request);
```
Comment (ayeshLK) [inline README.md]:
```suggestion
    ImageResponse response = check openaiImages->/images/generations.post(request);
```

File: README.md [hunk 8]
```diff
@@ -8,19 +8,97 @@
...
+
+    if (response is ImagesResponse) {
+        io:println("Created images: ", response.data);
+    } else {
+        io:println("Error: ", response);
+    }
+
```
Comment (ayeshLK) [inline README.md]:
```suggestion
    io:println("Created images: ", response.data);
```

File: README.md [hunk 9]
```diff
@@ -8,19 +8,97 @@
...
+### Step 2: Instantiate a new connector
+
+1. Create a `Config.toml` file and, configure the obtained credentials in the above steps as follows:
+
+```bash
+apiKey = "<API Key>"
+```
+
+2. Create a `images:ConnectionConfig` with the obtained API Key and initialize the connector with it.
```
Comment (ayeshLK) [inline README.md]:
```suggestion
Create a `images:ConnectionConfig` with the obtained API Key and initialize the connector.
```

Comment (G5andeepD) [reply README.md]: I was following the README of the twitter connector on this. 
https://github.com/ballerina-platform/module-ballerinax-twitter/blob/main/README.md

Comment (G5andeepD) [reply README.md]: ### Step 2: Instantiate a new connector

1. Create a `Config.toml` file and, configure the obtained credentials in the above steps as follows:

```bash
token = "<Access Token>"
```

2. Create a `twitter:ConnectionConfig` with the obtained access token and initialize the connector with it.

```ballerina
configurable string token = ?;

final twitter:Client twitter = check new({
    auth: {
        token
    }
});
```

Comment (ayeshLK) [reply README.md]: No need to have two steps. The suggestion is enough.

Comment (G5andeepD) [reply README.md]: oh ok. I will reflect the changes in module.md and package.md as well.

**Review by ayeshLK — APPROVED** [review]

**Review by ayeshLK — DISMISSED** [review]

File: README.md [hunk 10]
```diff
@@ -8,19 +8,85 @@
...
+    images:CreateImageRequest request = { 
+        prompt: "A cute baby sea otter",        
+        model: "dall-e-3",
+        n: 1,
+        size: "1024x1024",
+        response_format: "url"        
+    };
+
+    ImageResponse response = check openaiImages->/images/generations.post(request);
+    io:println("Created images: ", response.data);
```
Comment (ayeshLK) [inline README.md]:
```suggestion
```

**Review by NipunaRanasinghe — APPROVED** [review]

**Pull request merged**